### PR TITLE
Add CustomDashboardThemeConfigProvider component

### PR DESCRIPTION
**Pull request by RAVEENSR**
## Purpose
Add CustomDashboardThemeConfigProvider component and reorganise apim dashboard components.

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

**Comment by RAVEENSR** [comment]
Merge this PR after releasing carbon dashboards 4.1.14

File: components/org.wso2.analytics.apim.dashboards/org.wso2.analytics.apim.dashboards.theme.config.provider/src/main/java/org/wso2/analytics/apim/dashboards/theme/config/provider/CustomDashboardThemeConfigProvider.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    }
+
+    protected void unsetConfigProvider(ConfigProvider configProvider) {
+        LOGGER.debug("An instance of class '{}' unregistered as a config provider.",
+                configProvider.getClass().getName());
+    }
+
+    @Override
+    public String getPath(String username) throws DashboardException {
+        String themeConfigResourcesPath = this.dashboardConfigurations.getThemeConfigResourcesPath();
```
Comment (TanyaM) [inline components/org.wso2.analytics.apim.dashboards/org.wso2.analytics.apim.dashboards.theme.config.provider/src/main/java/org/wso2/analytics/apim/dashboards/theme/config/provider/CustomDashboardThemeConfigProvider.java]: With this, somebody who doesn't need to customize the theme, but want to change the hostname will also need to have this configuration. Isn't it?

Comment (RAVEENSR) [reply components/org.wso2.analytics.apim.dashboards/org.wso2.analytics.apim.dashboards.theme.config.provider/src/main/java/org/wso2/analytics/apim/dashboards/theme/config/provider/CustomDashboardThemeConfigProvider.java]: In that case DefaultDashboardThemeConfigProvider will handle.
https://github.com/wso2/carbon-dashboards/blob/master/components/dashboards/org.wso2.carbon.dashboards.core/src/main/java/org/wso2/carbon/dashboards/core/DefaultDashboardThemeConfigProvider.java

Comment (TanyaM) [reply components/org.wso2.analytics.apim.dashboards/org.wso2.analytics.apim.dashboards.theme.config.provider/src/main/java/org/wso2/analytics/apim/dashboards/theme/config/provider/CustomDashboardThemeConfigProvider.java]: Got it. Just saw https://github.com/wso2/analytics-apim/pull/950/files#diff-ff1fb90a787eacb45655c8c16c3edab1R40

**Pull request merged**